workers/unity/Assets/Playground/Config/PlayerTemplate.cs

```diff
@@ -40,14 +36,14 @@ public static class PlayerTemplate
                 .SetPersistence(false)
                 .SetReadAcl(WorkerUtils.AllWorkerAttributes)
                 .SetEntityAclComponentWriteAccess(WorkerUtils.UnityGameLogic)
-                .AddComponent(transform, clientAttribute)
                 .AddComponent(playerInput, clientAttribute)
                 .AddComponent(launcher, WorkerUtils.UnityGameLogic)
-                .AddComponent(clientHeartbeat, clientAttribute)
-                .AddComponent(serverHeartbeat, WorkerUtils.UnityGameLogic)
                 .AddComponent(score, WorkerUtils.UnityGameLogic)
                 .AddComponent(cubeSpawner, WorkerUtils.UnityGameLogic);
 
+            TransformSynchronizationEntityBuilderHelper.AddComponents(entityBuilder, clientAttribute);
+            PlayerLifecycleEntityBuilderHelper.AddComponents(entityBuilder, clientAttribute, WorkerUtils.UnityGameLogic);
+
             return entityBuilder.Build();
```

Comment:
If the `entityBuilder` is somehow made immutable (because builder patterns make that trivial) then the helper.AddComponents could potentially be no-op at this point. See my comment on them returning the entity builder.

That way:

```c#
entityBuilder = TransformSynchronizationEntityBuilderHelper.AddComponents(entityBuilder, clientAttribute);
entityBuilder = PlayerLifecycleEntityBuilderHelper.AddComponents(entityBuilder, clientAttribute, WorkerUtils.UnityGameLogic);

return entityBuilder.Build();
```